File: 448-number-disappeared.rb
def find_disappeared_numbers(nums)
    result = []
    nums.each do |el|
        index = el.abs - 1
        if (nums[index] > 0)
            nums[index] = - nums[index]
        end
    end
    nums.each_with_index do |el, idx|
        if el > 0
            result << (idx + 1)
        end
    end
    result
end

File: 448-number-disappeared.js
var findDisappearedNumbers = function(nums) {
    let result = [];
    for (let i = 0; i < nums.length; i++) {
        let index = Math.abs(nums[i]) - 1;
        if (nums[index] > 0) {
            nums[index] = - nums[index];
        }
    }
    for (let j = 0; j < nums.length; j++) {
        if (nums[j] > 0) result.push(j + 1);
    }
    return result;
};
